### ISSUE TITLE
Alternate algorithms for finding d-separation

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#1066]
### Subject of the issue
The current implementation of finding d-connected nodes is based on [1] which is great when we want to find all the nodes which are d-connected to some given node. But is slow in the case when we want to just check if two nodes are d-connected or not. [2] and [3] propose faster algorithms (O(|E|) and O(|V|^2) respectively) for checking whether two nodes are d-connected. Implement one of these algorithms to check if d-separation between two given nodes.

[1] (Algorithm 3.1 page 84) Koller, D., Friedman, N., & Bach, F. (2009). Probabilistic graphical models: principles and techniques. MIT press.
[2] Geiger, D., Verma, T., & Pearl, J. (1990). d-separation: From theorems to algorithms. In Machine Intelligence and Pattern Recognition (Vol. 10, pp. 139-148). North-Holland.
[3] Lauritzen, S. L., Dawid, A. P., Larsen, B. N., & Leimer, H. G. (1990). Independence properties of directed Markov fields. Networks, 20(5), 491-505.

### Your environment
* pgmpy version: All
* Python version: All
* Operating System: All

**Comment by nags0x** [comment]
@ankurankan , I will give it a shot /// will spend required time on getting intuition on [1,2,3] & then proceed further...

**Comment by ankurankan** [comment]
@nags0x There is already an open PR for this.

**Comment by satvshr** [comment]
@ankurankan Current active PR implements index [2] out of the 3 mentioned above, so can I try implementing one of the other 2? If yes should I implement it in the `is_dconnected` function and remove the current implementation (which is what I understood from a quick glance at the active PR for this issue)

**Comment by ankurankan** [comment]
@satvshr No, we shouldn't implement anything else for this issue. We only needed one algorithm for this.
